pex/pep_427.py

```diff
@@ -282,34 +282,49 @@ def record_files(
 
     dist = Distribution(location=dest, metadata=wheel.dist_metadata())
     entry_points = dist.get_entry_map()
-    for entry_point in itertools.chain.from_iterable(
+    for named_entry_point in itertools.chain.from_iterable(
         entry_points.get(key, {}).values() for key in ("console_scripts", "gui_scripts")
     ):
-        script_abspath = os.path.join(install_paths.scripts, entry_point.name)
-        with safe_open(script_abspath, "w") as fp:
-            fp.write(
-                dedent(
-                    """\
-                    {shebang}
-                    # -*- coding: utf-8 -*-
-                    import importlib
-                    import sys
-
-                    object_ref = "{object_ref}"
-                    modname, qualname_separator, qualname = object_ref.partition(':')
-                    entry_point = importlib.import_module(modname)
-                    if qualname_separator:
-                        for attr in qualname.split('.'):
-                            entry_point = getattr(entry_point, attr)
```

Comment:
The re-factoring of `{EntryPoint,CallableEntryPoint}` -> `NamedEntryPoint <<containing>> {ModuleEntryPoint,CallableEntryPoint}` highlighted that this generated console script was wrong for modules (i.e.: when this if branch is not taken) since a module is not a callable! It turns out that in practice, console script entrypoints are always functions (callables) and not modules and so no bug reports. That said, I switched Pex's own `pex` entrypoint from a function to the `pex` module in its own `pyproject.toml` and verified that the pex wheel still builds fine and then a Pex PEX venv built from that wheel had a `pex` console script that then failed. Thus, the fix.